worlds/shorthike/__init__.py

```diff
@@ -0,0 +1,155 @@
+        self.multiworld.itempool += [self.create_item("Silver Feather") for _ in range(self.options.silver_feathers)]
+        self.multiworld.itempool += [self.create_item("Bucket") for _ in range(self.options.buckets)]
+
+    def create_regions(self) -> None:
+        menu_region = Region("Menu", self.player, self.multiworld)
+        self.multiworld.regions.append(menu_region)
+        
+        main_region = Region("Hawk Peak", self.player, self.multiworld)
+
+        for loc in self.location_name_to_id.keys():
+            main_region.locations.append(ShortHikeLocation(self.player, loc, self.location_name_to_id[loc], main_region))
+
+        self.multiworld.regions.append(main_region)
+
+        menu_region.connect(main_region)
+
+        if self.options.goal == 0:
+            # Nap
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7)))
+        elif self.options.goal == 1:
+            # Photo
+            self.multiworld.completion_condition[self.player] = lambda state: state.has("Golden Feather", self.player, 12)
+        elif self.options.goal == 2:
+            # Races
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7)))
+        elif self.options.goal == 3:
+            # Help Everyone
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, 12)
+                and state.has("Toy Shovel", self.player) and state.has("Camping Permit", self.player)
+                and state.has("Motorboat Key", self.player) and state.has("Headband", self.player)
+                and state.has("Wristwatch", self.player) and state.has("Seashell", self.player, 15)
+                and state.has("Shell Necklace", self.player))
+        elif self.options.goal == 4:
+            # Fishmonger
+            self.multiworld.completion_condition[self.player] = lambda state: (state.has("Golden Feather", self.player, get_min_feathers(self, 7, 9))
+                or (state.has("Bucket", self.player) and state.has("Golden Feather", self.player, 7))
+                and state.has("Fishing Rod", self.player))
+
+    def set_rules(self):
+        create_rules(self, location_table)
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        locations: Dict[int, Any] = {}
+
+        multiworld = self.multiworld
+        player = self.player
+        options = self.options
+
+        for loc in multiworld.get_filled_locations(player):
+            if loc.item.code is None:
+                continue
+            else:
+                data = {
```

Comment:
Except for chest angle everything here can be gotten via LocationScout, which is preferred as it reduces the multidata filesize.

Reply:
And chest angle seems to be static anyway? Might as well bake it into the mod instead of  storing it on the server and tranferring it?

Reply:
I went ahead and removed this from the slot data entirely. The apworld should be good to go, although I am still awaiting these changes to be made on the client mod itself. I'll let you know once that change is made.

Reply:
@Berserker66 We can remove the item name and the player name from the slot data since those only matter once the player collects the item.
The type is used for each location to display a golden chest instead of a regular one if the item is progression.  This needs to be known right after connection before loading the gameplay scene.  This isn't necessary, but is a cool feature so we can remove that also if it is a dealbreaker.
However, I don't want to hardcode the ap ids into the mod, so can we at least leave the mapping of ap_ids to internal_ids?  The huge benefit of this is that whenever we add new locations or modify the ids, players don't have to update their client, as all that data is stored in only one place, the apworld.
If leaving the { internal_id, ap_id } in the list of locations in slot data is alright and works in the mod, I suppose I can also update blasphemous (Which uses this same method) to remove the extra info in slot data there as well to save even more space.

Reply:
should just be able to use NetworkItem.flags instead https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#networkitem
which should be in the scouts as well

Reply:
This will be moved to the client :+1: